### PR TITLE
Introduce missing button element classes for the account orders and downloads

### PR DESCRIPTION
**Pull request by xristos3490**
### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

Before starting ensure that you are using a block theme like TT2, or TT3.

**Testing in order-again context:**

1. Create a new order and get to the Order Received page (don't close it)
2. Process the order from wp-admin and refresh the previous page to see the "Order again" button.
3. Notice that it's missing some styles. See https://d.pr/i/oVXIas
4. Apply this fix, and ensure the button appears correctly.

**Testing in the pagination buttons on the account page:**
1. Navigate to My Account > Orders page and ensure that you have enough so the pagination shows up _(hint: one could use the `woocommerce_my_account_my_orders_query` filter to wire a small number (e.g., 2) in the `limit` key for the query)_
2. Notice the buttons are missing styles. See https://d.pr/i/lYs1jg
3. Apply this fix, and ensure the buttons appear correctly.

**Testing in the no-downloads cta button on the account page:**
1. Navigate to My Account > Downloads page and ensure your user hasn't downloaded yet.
2. Check the CTA button inside the notice saying "No downloads available yet"; it shouldn't look right.
3. Apply this patch and ensure the CTA shows up correctly.

### CODE REVIEW

**Comment by xristos3490** [comment]
Hey @coreymckrill! fyi I left the template bumps so we can decide on which version these will land. I can then push another commit containing the proper version number.

File: plugins/woocommerce/includes/wc-template-functions.php [hunk 1]
```diff
@@ -2747,6 +2747,7 @@ function woocommerce_order_again_button( $order ) {
 			'order/order-again.php',
 			array(
 				'order'           => $order,
+				'wp_button_class' => wc_wp_theme_get_element_class_name( 'button' ) ? ' ' . wc_wp_theme_get_element_class_name( 'button' ) : '',
```
Comment (coreymckrill) [inline plugins/woocommerce/includes/wc-template-functions.php:2755]:
```suggestion
				'wp_button_class' => wc_wp_theme_get_element_class_name( 'button' ),
```

Optional, non-blocking suggestion, but I would argue for having this template parameter only contain the classname, and worry about the whitespace in the template itself. And also, probably not worry about extra whitespace at the end of the `class` attribute in the case that `wp_button_class` is empty.

File: plugins/woocommerce/templates/order/order-again.php [hunk 2]
```diff
@@ -12,12 +12,12 @@
...
- * @version 3.5.0
+ * @version x.x.x
  */
 
 defined( 'ABSPATH' ) || exit;
 ?>
 
 <p class="order-again">
-	<a href="<?php echo esc_url( $order_again_url ); ?>" class="button"><?php esc_html_e( 'Order again', 'woocommerce' ); ?></a>
+	<a href="<?php echo esc_url( $order_again_url ); ?>" class="button<?php esc_attr( $wp_button_class ); ?>"><?php esc_html_e( 'Order again', 'woocommerce' ); ?></a>
```
Comment (coreymckrill) [inline plugins/woocommerce/templates/order/order-again.php]:
```suggestion
	<a href="<?php echo esc_url( $order_again_url ); ?>" class="button<?php echo esc_attr( $wp_button_class ); ?>"><?php esc_html_e( 'Order again', 'woocommerce' ); ?></a>
```

**Review by coreymckrill — CHANGES_REQUESTED** [review]
@xristos3490 👍 looks good, other than the needed fix on the order again template 😅 

This PR also needs a rebase since it was originally submitted before the monorepo upgraded pnpm to v8.

File: plugins/woocommerce/includes/wc-template-functions.php (hunk 1 above)
Comment (xristos3490) [reply plugins/woocommerce/includes/wc-template-functions.php:2755]: I was inspired by other instances of the same fix throughout the codebase (like [here](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/wc-template-functions.php#L3247)) 

To be honest, I also believe that the templates should be responsible for managing the classes. Ideally, I'd expect to _echo_ them using arrays and the `implode` function. We could move forward with this as _is_, progress with other changes in regards to the _block-ification_ of the templates, and then return to create a more organized solution. 

I'd be happy to change it, though. Thoughts?

**Comment by xristos3490** [comment]
Thanks for the review, @coreymckrill! I've updated the base of this fix.

File: plugins/woocommerce/includes/wc-template-functions.php (hunk 1 above)
Comment (coreymckrill) [reply plugins/woocommerce/includes/wc-template-functions.php:2755]: > Ideally, I'd expect to echo them using arrays and the implode function. We could move forward with this as is, progress with other changes in regards to the block-ification of the templates, and then return to create a more organized solution.

👍 Yeah, doing this separately sounds like the right approach. No need to make this PR more complicated.

**Review by coreymckrill — APPROVED** [review]

**Review by coreymckrill — CHANGES_REQUESTED** [review]
Bah, I approved this and then realized that the code sniff test is [failing](https://github.com/woocommerce/woocommerce/actions/runs/4893494458/jobs/8736522628?pr=37933). Should be good to go after that's fixed.

**Comment by xristos3490** [comment]
Nice catch, @coreymckrill! 🙇 

I have made some adjustments to ensure PHPCS is satisfied. However, we still need to determine the appropriate version number for the template modifications. What do you think? Would 7.7.0 be a suitable choice, or perhaps 7.6.2?

**Review by coreymckrill — DISMISSED** [review]
> we still need to determine the appropriate version number for the template modifications. What do you think? Would 7.7.0 be a suitable choice, or perhaps 7.6.2?

This doesn't seem like an urgent bug fix that would qualify for a point release (let me know if you disagree). Given that 7.7.0 is now out (sorry, was at a conference), I think 7.8.0 would be the right target here.

**Comment by xristos3490** [comment]
> I think 7.8.0 would be the right target here.

Totally agree! Changing templates is always a good idea to do it, at least, on a minor release! Will update the headers shortly!

**Comment by xristos3490** [comment]
Updated the template version and rebased the branch. 

Two templates have already been upgraded to 7.8.0. So, the `Highlight templates changes` is probably a false positive.

**Review by coreymckrill — APPROVED** [review]

**Comment by coreymckrill** [comment]
Thanks @xristos3490 !

**Pull request merged**